devito/passes/iet/definitions.py

```diff
@@ -131,37 +146,7 @@ def place_definitions(self, iet, **kwargs):
                     # E.g., a generic SymPy expression
                     pass
 
-        # Place symbols in a memory space
-        if not iet.is_ElementalFunction:
-            writes = set()
-            reads = set()
-            for efunc in kwargs.get('efuncs', []):
-                for i in FindNodes(Expression).visit(efunc):
-                    if i.write.is_Function:
-                        writes.add(i.write)
-                    reads = (reads | {r for r in i.reads if r.is_Function}) - writes
-
-            for i in filter_sorted(writes):
-                self._map_function_on_high_bw_mem(i, storage)
-            for i in filter_sorted(reads):
-                self._map_function_on_high_bw_mem(i, storage, read_only=True)
-
-        # Introduce symbol definitions going in the low latency memory
-        mapper = dict(storage._on_low_lat_mem)
-        iet = Transformer(mapper, nested=True).visit(iet)
-
-        # Introduce symbol definitions going in the high bandwidth memory
-        header = []
-        footer = []
-        for decl, alloc, free in storage._on_high_bw_mem:
-            if decl is None:
-                header.append(alloc)
-            else:
-                header.extend([decl, alloc])
-            footer.append(free)
-        if header or footer:
-            body = List(header=header, body=iet.body, footer=footer)
-            iet = iet._rebuild(body=body)
+        iet = self._dump_storage(iet, storage)
```

Comment:
note for reviewers: I simply moved this to a method so that it can be called by subclasses when needed